### PR TITLE
Validation Maximum Annotation

### PR DESCRIPTION
**Pull request by Schmarvinius**
# Enhanced File Attachment Validation with Maximum Size Annotation

### New Features

✨ Introduced file size validation using the `@Validation.Maximum` annotation to enforce maximum attachment sizes during upload operations. The feature supports human-readable size formats (e.g., "20MB", "5MB") and provides proper HTTP 413 error responses when size limits are exceeded.

#### Core Implementation

* `UpdateAttachmentsHandler.java`: Enhanced update handler to validate attachment sizes before processing. Refactored attachment deletion logic to use direct iteration instead of `CdsDataProcessor`. Improved code formatting and field ordering for better maintainability.

* `FileSizeUtils.java`: Created utility class to parse human-readable file size strings into byte values. Supports both decimal (KB, MB, GB, TB) and binary (KiB, MiB, GiB, TiB) units with comprehensive validation and error handling.

* `ModifyApplicationHandlerHelper.java`: Integrated size validation logic using the `getValMaxValue()` method to extract annotation values. Added Content-Length header validation and proper error handling with HTTP 413 responses when limits are exceeded.

* `ExtendedErrorStatuses.java`: Added new enum defining `CONTENT_TOO_LARGE` error status with HTTP 413 code for size validation failures.

* `CountingInputStream.java`: New class that wraps input streams and tracks bytes read. Throws `ServiceException` when the configured size limit is exceeded during streaming, enabling real-time validation during file upload.

* `ApplicationHandlerHelper.java`: Added `extractKeys()` helper method to extract key fields from CdsData based on entity definitions.

* `CreateAttachmentsHandler.java`: Updated to wrap content streams with `CountingInputStream` for size validation. Added `restoreError()` method to handle HTTP 413 errors with contextual information about size limits.

* `messages.properties`: Added localized error message `AttachmentSizeExceeded` for attachment size validation failures.

#### Testing

* `CreateAttachmentsHandlerTest.java`, `UpdateAttachmentsHandlerTest.java`, `DraftPatchAttachmentsHandlerTest.java`: Updated tests to mock `ParameterInfo` and handle validation scenarios. Improved test assertions and data setup for better coverage. Enhanced to capture and verify `CountingInputStream` instances in the processing flow.

* `FileSizeUtilsTest.java`: Comprehensive test suite for file size parsing covering valid formats, case-insensitive units, error handling, and edge cases.

* `ExtendedErrorStatusesTest.java`: Test coverage for the new error status enum.

* `SizeLimitedAttachmentValidationNonDraftTest.java`, `SizeLimitedAttachmentsSizeValidationDraftTest.java`: Integration tests validating the size limit enforcement for both draft and non-draft scenarios, including upload and update operations.

* `OdataRequestValidationBase.java`, `DraftOdataRequestValidationBase.java`: Made `persistenceService` field protected and added helper methods for sizeLimitedAttachments navigation and content upload.

* `CountingInputStreamTest.java`: Added comprehensive tests for `CountingInputStream`, covering read operations (single byte, byte arrays, skip), limit checking, error handling, and edge cases.

* `ModifyApplicationHandlerHelperTest.java`: Added tests for size validation logic, including Content-Length header validation and streaming size checks.

* `MarkAsDeletedAttachmentEventTest.java`: Added tests to verify path handling in the deletion flow, including null path scenarios and cases with mismatched content IDs.

#### Sample Application

* `attachments.cds`: Demonstrated usage by adding `@Validation.Maximum: '20MB'` annotation to Books attachments content field in the bookshop sample. Added `sizeLimitedAttachments` composition with a `@Validation.Maximum: '5MB'` annotation for testing purposes.

* `db-model.cds`: Updated Items entity to use composition of `Attachment` directly without `parentKey`, added `sizeLimitedAttachments` and `defaultSizeLimitedAttachments` to EventItems for testing scenarios, and added corresponding annotations for size limits.

* `data-model.cds`: Added `sizeLimitedAttachments` composition to Roots entity for integration testing purposes.

* `test-service.cds`: Applied `@Validation.Maximum: '5MB'` annotation to `Roots.sizeLimitedAttachments` for validation testing.

* `pom.xml`: Updated project version from `1.2.4` to `1.2.5-SNAPSHOT` and upgraded Spring Boot to `3.5.9`.

* `.gitignore`: Added `.cdsrc-private.json` to ignore CDS-specific private configuration files.

#### Documentation

* `README.md`: Added new section "Specify the maximum file size" documenting how to use the `@Validation.Maximum` annotation on attachment content properties to enforce file size limits. Included example code and clarified the default limit of 400MB.

- [ ] 🔄 Regenerate and Update Summary

### CODE REVIEW

**Comment by Schmarvinius** [comment]
update readme

File: cds-feature-attachments/src/main/java/com/sap/cds/feature/attachments/handler/applicationservice/helper/ModifyApplicationHandlerHelper.java [hunk 1]
```diff
@@ -64,12 +74,54 @@ public static InputStream handleAttachmentForEntity(
     ReadonlyDataContextEnhancer.restoreReadonlyFields((CdsData) path.target().values());
     Attachments attachment = getExistingAttachment(keys, existingAttachments);
     String contentId = (String) path.target().values().get(Attachments.CONTENT_ID);
+    String contentLength = eventContext.getParameterInfo().getHeader("Content-Length");
+    String maxSizeStr = getValMaxValue(path.target().entity(), existingAttachments);
+    eventContext.put(
+        "attachment.MaxSize",
+        maxSizeStr); // make max size available in context for error handling later
+    ServiceException TOO_LARGE_EXCEPTION =
```
[the remark this answers is not in the document]
Comment (Schmarvinius) [reply cds-feature-attachments/src/main/java/com/sap/cds/feature/attachments/handler/applicationservice/helper/ModifyApplicationHandlerHelper.java]: Rule of Three applies here

File: cds-feature-attachments/src/main/java/com/sap/cds/feature/attachments/handler/applicationservice/helper/FileSizeUtils.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    Matcher m = SIZE.matcher(input);
+    if (!m.matches()) {
+      throw new IllegalArgumentException("Invalid size: " + input);
+    }
+    BigDecimal value = new BigDecimal(m.group(1));
+    String unitRaw = m.group(2); // Cannot be null due to * quantifier in regex
+    String unit = unitRaw.toUpperCase();
+
+    // if (unit.length() == 1) unit = unit + "B"; // for people using K instead of KB
+    Long mul = MULTIPLIER_TO_BYTES.get(unit);
```
[the remark this answers is not in the document]
Comment (Schmarvinius) [reply cds-feature-attachments/src/main/java/com/sap/cds/feature/attachments/handler/applicationservice/helper/FileSizeUtils.java]: YAGNI applies here. And I think MULTIPLIER_TO_BYTES is as descriptive as it can get.
